nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_constructors.rs

```diff
@@ -633,3 +637,147 @@ impl TypedArrayPrototypes {
+            match first_argument {
+                TypedArray::Int8Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, i8>(agent, o, first_argument)?
+                }
+                TypedArray::Uint8Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, u8>(agent, o, first_argument)?
+                }
+                TypedArray::Uint8ClampedArray(_) => initialize_typed_array_from_typed_array::<
+                    T,
+                    U8Clamped,
+                >(agent, o, first_argument)?,
+                TypedArray::Int16Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, i16>(agent, o, first_argument)?
+                }
+                TypedArray::Uint16Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, u16>(agent, o, first_argument)?
+                }
+                TypedArray::Int32Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, i32>(agent, o, first_argument)?
+                }
+                TypedArray::Uint32Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, u32>(agent, o, first_argument)?
+                }
+                TypedArray::BigInt64Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, i64>(agent, o, first_argument)?
+                }
+                TypedArray::BigUint64Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, u64>(agent, o, first_argument)?
+                }
+                TypedArray::Float32Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, f32>(agent, o, first_argument)?
+                }
+                TypedArray::Float64Array(_) => {
+                    initialize_typed_array_from_typed_array::<T, f64>(agent, o, first_argument)?
+                }
+            }
+        } else if let Ok(first_argument) = ArrayBuffer::try_from(first_argument) {
+            // iii. Else if firstArgument has an [[ArrayBufferData]] internal slot, then
+            // 1. If numberOfArgs > 1, let byteOffset be args[1]; else let byteOffset be undefined.
+            let byte_offset = if arguments.len() > 1 {
+                Some(arguments.get(1))
+            } else {
+                None
+            };
+
+            // 2. If numberOfArgs > 2, let length be args[2]; else let length be undefined.
+            let length = if arguments.len() > 2 {
+                Some(arguments.get(2))
+            } else {
+                None
+            };
+
+            // 3. Perform ? InitializeTypedArrayFromArrayBuffer(O, firstArgument, byteOffset, length).
+            initialize_typed_array_from_array_buffer::<T>(
+                agent,
+                gc.reborrow(),
+                o,
+                first_argument,
+                byte_offset,
+                length,
+            )?;
+        }
```

Comment:
**issue:** Seems to be missing the `else {}` branch and instead this gets done to TA and AB initialized TA's as well.